CoroutineTemplate/domain/src/main/java/co/nimblehq/coroutine/repository/UserRepository.kt

```diff
@@ -0,0 +1,8 @@
+package co.nimblehq.coroutine.repository
+
+import co.nimblehq.coroutine.response.UserResponse
+
+interface UserRepository {
+
+    suspend fun getUsers(): List<UserResponse>
```

Comment:
`Repository` should have only `domain` reference and `RepositoryImpl` should do the mappings (`data` models to `domain` models).
```suggestion
    suspend fun getUsers(): List<User>
```

Reply:
@AVI5HEK If we do that, we have to `try/cach`  and return the wrap Result inside `RepositoryImpl` as my previous implementation?